### PR TITLE
Improved errors handling

### PR DESCRIPTION
**Pull request by shegeley**
I'm new to Clojure and started to use it for personal project connected with Aritable. For this reason I improved a library a little to handle 429 errors (too many requests per minute) and pagination (retrieving more than 100 records with one request). Please check the code before merging: as I said I'm newbie.

### CODE REVIEW

File: src/airtable_clj/core.clj [hunk 1]
```diff
@@ -53,7 +51,7 @@
                               :multi-param-style :array}
                        (seq query-params) (assoc :query-params query-params))
         response (http/get url http-options)
-        _ (handle-api-error response)
+        _ (handle-api-error response select options)
```
Comment (shegeley) [inline src/airtable_clj/core.clj]: I'm not sure how does 
```clojure
        _ (handle-api-error response select options)
```
works. It uses an underscore, means ignores other result (?).

File: .gitignore [hunk 2]
```diff
@@ -1,11 +0,0 @@
-/target
```
Comment (EvanHahn) [inline .gitignore:1]: Why was this file removed? (`git checkout master .gitignore` should restore it if this was removed by accident.)

File: README.md [hunk 3]
```diff
@@ -1,18 +1,12 @@
-# airtable-clj
```
Comment (EvanHahn) [inline README.md:1]: This file doesn't seem like it should've been changed as part of this pull request. Could you revert?

File: project.clj [hunk 4]
```diff
@@ -1,11 +1,12 @@
-(defproject airtable-clj "0.1.0-SNAPSHOT"
```
Comment (EvanHahn) [inline project.clj:1]: This file doesn't seem like it should've been changed as part of this pull request. Could you revert?

**Review by EvanHahn — CHANGES_REQUESTED** [review]
It's been a long time since I've worked on this library (and on anything in Clojure)! Thanks for taking a look.

I've left a partial review with a few things to clean up before taking a more thorough look.

Thanks again!

**Comment by shegeley** [comment]
Thank you for a review. Unfortunately I'm new Github and open source contributions. How do I revert lines that you've commented? I can't find that function anywhere in Github web interface. Should I just make changes in given files on my local machine and push them to the branch of pull request?

**Comment by shegeley** [comment]
Also please take a look at that part https://github.com/EvanHahn/airtable-clj/pull/1#discussion_r387509695 I've commented.

File: src/airtable_clj/core.clj (hunk 1 above)
Comment (EvanHahn) [reply src/airtable_clj/core.clj]: Yes—it means that the result is ignored. This will `throw` an error if there was a response error.

**Comment by EvanHahn** [comment]
You'll do those reversions as part of your local Git branch, and then you'll push those up to GitHub.

You'll do something like this:

```sh
# Make sure you're on the right branch
git checkout improved-errors-handling

# Revert things
git checkout master project.clj .gitignore

# Commit those changes
git commit

# Push them up to your branch as normal, which will update this pull request
git push
```

**Comment by EvanHahn** [comment]
Shall I take another look? Or are you still working?

**Comment by shegeley** [comment]
Yes. Take a look please.

File: README.md (hunk 3 above)
Comment (EvanHahn) [reply README.md:1]: This comment still applies.

File: .gitignore (hunk 2 above)
Comment (EvanHahn) [reply .gitignore:1]: This comment still applies.

File: project.clj (hunk 4 above)
Comment (EvanHahn) [reply project.clj:1]: This comment still applies.

File: src/airtable_clj/core.clj [hunk 5]
```diff
@@ -1,11 +1,12 @@
 (ns airtable-clj.core
   (:require [clj-http.client :as http]
             [cheshire.core :as json]
+            [debux.core :as debux]
```
Comment (EvanHahn) [inline src/airtable_clj/core.clj:4]: Is this import used anywhere? It doesn't look like it...

**Review by EvanHahn — CHANGES_REQUESTED** [review]
Did a quick review and left a few more comments (looks like not everything was addressed from my comments before).

Let me know if I can help with anything!

**Comment by EvanHahn** [comment]
There's been no movement on this PR for over 2 years, so I'm going to close. Let me know if that's wrong!
